independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java

```diff
@@ -49,6 +49,11 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
                 priority = annotation.value().asInt();
                 priorityDeclared = true;
             }
+            ScopeInfo scopeAnnotation = beanDeployment.getScope(annotation.name());
+            if (scopeAnnotation != null && !BuiltinScope.DEPENDENT.is(scopeAnnotation)) {
+                throw new DefinitionException(
```

Comment:
Speaking of what decorator can and cannot do - what about decorators for `BeanManager` type? Specification [forbids that](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#decorators) as well; should we make sure it doesn't work as part of this PR?

Reply:
Hm, I think that `BeanManager` is not even injectable in ArC, i.e. there is no built-in bean of that type and so you won't be able to decorate it unless you explicitly produce it.

Reply:
Oh, I automatically assumed that we have such bean. My bad, forget I said anything :)